src/reducers/sources.js

```diff
@@ -640,32 +528,34 @@ export function getProjectDirectoryRoot(state: OuterState): string {
   return state.sources.projectDirectoryRoot;
 }
 
-function getAllRelativeSources(state: OuterState): SourcesMapByThread {
-  return state.sources.relativeSources;
+function getAllDisplayedSources(state: OuterState) {
+  return state.sources.displayed;
 }
 
 function getChromeAndExtenstionsEnabled(state: OuterState) {
   return state.sources.chromeAndExtenstionsEnabled;
 }
 
-export const getRelativeSources: GetRelativeSourcesSelector = createSelector(
+export const getDisplayedSources: GetDisplayedSourcesSelector = createSelector(
+  getSources,
   getChromeAndExtenstionsEnabled,
-  getAllRelativeSources,
-  (chromeAndExtenstionsEnabled, relativeSources) => {
-    if (!chromeAndExtenstionsEnabled) {
-      return mapValues(relativeSources, threadSources => {
-        return omitBy(threadSources, source => source.isExtension);
-      });
-    }
-    return relativeSources;
+  getAllDisplayedSources,
+  (sources, chromeAndExtenstionsEnabled, displayed) => {
+    return mapValues(displayed, threadSourceIds => {
+      const threadSources = mapValues(threadSourceIds, (_, id) => sources[id]);
+      if (chromeAndExtenstionsEnabled) {
+        return threadSources;
+      }
+      return omitBy(threadSources, source => source.isExtension);
```

Comment:
we currently do not change the store value when the pref changes. I filed this issue for updating the logic and i think we can re-compute the displayed sources at that point too. 

https://github.com/firefox-devtools/debugger.html/issues/7981

the primary thing i'm concerned here is that this list will still re-compute when sources change like a source is loaded or something, so it is nice to cut down on any of the work..

Reply:
Would having separate stores for attributes that are prone to change (isPrettyPrinted, isBlackBoxed) be a solution to avoid re-computing? 

For example, we have a `state.sources.blackBoxedSources` that stores source ids that we can reference when selecting which sources to blackbox

Reply:
Not sure. Feel fee investigating